### ISSUE TITLE
Need better search, enhanced gui & ide like code map

### ISSUE THREAD

**Issue by andrewpros** [skylot/jadx#74]
Hi.

First of all i want to say that this is not a issue, but a feature request , jadx works great, but working with the gui could be, a lot better with some easy tricks.

**Better search** - i would be a lot easier if there were full apk string search, for now you can only search objects, but not strings, you can search text in selected file, but not in the whole code of application, that would be great.

Of course you can Save All and search in those files, but it has its problems, later on that.

**Enhanced gui** - as part of search and quicker operations, there should be some jump to file options, with quick file chooser, and when you press a letter in code list it should jump to next file in current node with name that starts by that letter.

We could also have a better working sync of code list with actual viewed files.

**IDE like code map** - for now we can do a ctrl + click on a function or variable and jump to its declaration, this is really great, but why no to go further, add option to find where variable or function is referenced in the whole apk, just right click in code list or editor and choose find all references.

It can be extended with more detailed references later, like find what assigns or only access a variable etc.

When jumping to other parts of code, there should be also a back/forward ability, as of now in a big file if you ctrl + click jump to function you can't go back and need to search the previous place manually.

Back to Save All, this is why it is problematic, you can search whole app in notepad++, but after finding the right file, analyzing it is problematic, you can't ctrl + click, you can go back to jadx gui, but there you need to find that class/function again, and it is not easy if you have 20 pages of classes to scroll in code list without any help in the gui, like proposed jump to file.

I think this suggestions would make jadx gui far better to work with it than now, especially then analyzing apks, those propositions are based on my work with jadx past month.

In my opinion they are easy to make and the pros are tremendous, jadx is already making some code map in memory, just add some functions to it.

I'm analyzing quite big apks, lots of classes, functions, and of top of that, they are also obfuscated, it is really hard now to find what you are looking when you have name like a, b c, aaa, bb, cc etc. all over the place, it is possible, but with those enhancements it would be a lot easier and faster.

**To sum it up, what i suggest**
- ability to jump to a file, maybe some shortcut, enter name, display file list (maybe real time), dbl click to jump or choose with arrows and enter
- jump to next file by pressing letter in code list
- better code list with code tabs sync, to make selected file always show in code list without problems when changing code tabs
- search in whole app code by string not only in selected file
- back/forward ability when jumping from one place to another
- find references, classes, functions, variables, the most valuable one in my opinion
- save jadx gui window size, it always resets
- maybe one more code enhancement, selection occurrences highlight

Hope at least some of this suggestions can be made.
